### ISSUE TITLE
Creating an optional popup for software checkmark

### ISSUE THREAD

**Issue by stephengroat** [2factorauth/twofactorauth#2067]
Looking at #2061, #2060, and #2057, it seems like if a specific software app is required, making the software checkmark popup a link to it would be a good idea (similar to the current exception system). It should probably be optional and need to have a list of the links to all of the different OS implementation (Android, iOS, etc.).

I'm envisioning the `_data` file looking like this:

```
  - name: Apple iCloud
    url: https://www.icloud.com
    img: icloud.png
    tfa: Yes
    sms: Yes
    software: Yes
      text: "iOS: www.apple.com"
    exceptions:
        text: "See http://support.apple.com/kb/HT5593 for a list of supported SMS carriers."
    doc: http://support.apple.com/kb/ht5570
```

and the website looking like it current

@psgs, any way you could help with this? I tried, but turned everything into spaghetti. I'm particularly having a hard time understanding what the `data.json` file does

**Comment by fpigerre** [comment]
Sure @stephengroat :smile: 
The `data.json` file allows machines to access data from the twofactorauth.org website using JSON. For example, a program could be written that would pull all the entries from https://twofactorauth.org/data.json and make a decision that would show people which website in a particular category provides the most comprehensive TFA solution.

**Comment by Carlgo11** [comment]
Let's wait with this for some time. First we need to decide if we're going to list things like this at all.

**Comment by kiwiant** [comment]
This would be great especially when someone like Apple forces users to use an iPhone for software tokens rather than standard authenticator software like the rest of the industry.

**Comment by peterbraden** [comment]
Indeed, there should be some way to differentiate between open OTP app usage, and specific app requirements. (Such as authy etc.)

**Comment by conorgil** [comment]
@stephengroat @psgs @Carlgo11 Have you given this any further thought? I think it would be incredibly useful to know which services require some specific app or proprietary solution, etc.

**Comment by Carlgo11** [comment]
I’d be open to the idea. Lots of people have requested a way to set what software services a site allows. But before we create some design for it I think the first step would be to try and create some sort of standard, list it in countributing.md & the wiki, create checks for it and then add it to data.json to start with.

**Comment by twilsonco** [comment]
+1 for specifying between generic software token (e.g. 2FA), and proprietary, e.g. Authy)

**Comment by Carlgo11** [comment]
Fixed with separate TOTP/Proprietary tags in v3.0
